docs/designers-developers/developers/block-api/block-registration.md

```diff
@@ -264,7 +264,7 @@ An object describing a variation defined for the block type can contain the foll
 - `attributes` (optional, type `Object`) – Values that override block attributes.
 - `innerBlocks` (optional, type `Array[]`) – Initial configuration of nested blocks.
 - `example` (optional, type `Object`) – Example provides structured data for the block preview. You can set to `undefined` to disable the preview shown for the block type.
-- `scope` (optional, type `string[]`) - the list of scopes where the variation is applicable. When not provided, it assumes all available scopes. Available options: `block`, `inserter`.
+- `scope` (optional, type `string[]`) - the list of scopes where the variation is applicable. When not provided, it assumes all available scopes. Available options: `block`, `inserter`, `transform`.
```

Comment:
Would be great to see further definition of these options, I understand `inserter` and `transform` would control whether the block is shown in the inserter or as a transform, but wasn't straight away sure what the `block` option does. 

Is that for something like the columns block where we see the variations in a placeholder?

Reply:
I have this in mind yes and will update docs where needed later. For the record:
- inserter - shows variation in inserter
- block - used by Placeholders like you mentioned in Columns
- transform - will be shown in the component for block variation transformations

I do also feel that this abstraction here https://github.com/WordPress/gutenberg/blob/b34ecc23f89066e3e7bb710136efa46e29ebe025/packages/blocks/src/store/selectors.js#L87 makes more than it should be and this surfaced with the addition of `transform` as `scope`.

>When not provided, it assumes all available scopes.

With `transform` we want to have this scope and doesn't fall in the rest cases that `no scope means every scope`.  What do you think about changing this to have an option like `strictFiltering` which will default to `false` for back compat?

Reply:
It is about time to convert this list into a proper JSDoc type and list all options according to the comment you shared. By the way, should it be plural `transforms` as in the target place where it's going to be used?

> > When not provided, it assumes all available scopes.
> With `transform` we want to have this scope and doesn't fall in the rest cases that no scope means every scope. What do you think about changing this to have an option like `strictFiltering` which will default to `false` for back compat?

Do we really need a new option there? Wouldn't it be enough to set the list of scopes to `[ 'block', `inserter` ]` if the scope param is not provided?